File: other/claude/FindTheFirstRepeatedElement.java
import java.util.Set;
import java.util.HashSet;

public class FindTheFirstRepeatedElement{

    private static int firstrepeatedElement(int[] nums) {

        if (nums == null || nums.length < 2) {
            return -1;
        }

        Set<Integer> seenElement = new HashSet<>();

        for (int i : nums) {

            if (!seenElement.add(i)) {
                return i;
            }
        }
        return -1;
    }

    public static void main(String[] args){

        int[] nums = new int[] {7,7};
        System.out.println(firstrepeatedElement(nums));
    }
}

File: other/claude/FindTheFirstRepeatedElement.js

const findFirstRepeatNumber = (nums) => {

    if (nums.length === 1) return nums[0];
    if (nums.length <= 0) return -1;

    let set = new Set();

    for (let i of nums){

        if (set.has(i)) return i;
        set.add(i);
    }
}

let nums = [1,2,34,653,2,4,34,9];
console.log(findFirstRepeatNumber(nums));